db/structure.sql

```diff
@@ -135,7 +135,8 @@ CREATE TABLE locks (
     created_at timestamp without time zone NOT NULL,
     updated_at timestamp without time zone NOT NULL,
     environment_id integer NOT NULL,
-    user_id integer NOT NULL
+    user_id integer NOT NULL,
+    waiting boolean DEFAULT false NOT NULL
```

Comment:
We should probably add a db level uniqueness constraint on this. A user can only have 1 "waiting" lock per environment:

```sql
CREATE UNIQUE INDEX one_waiting_lock_per_environment ON (locks) USING btree(environment_id, user_id, waiting) WHERE waiting = 't'
```